Cluster, replicaset, and instance names (#8289)
tarantool/tarantool: changelogs/unreleased/global-names.md

## feature/replication

> * New option `box.cfg.cluster_name` allows to assign a human-readable name to
>   the entire cluster with all its replicasets. It has to match in all instances
>   of all the replicasets in one cluster and is displayed in
>   `box.info.cluster.name` (gh-5029).

Reviewer: ```suggestion
* A new option `box.cfg.cluster_name` allows assigning a human-readable name
  to the entire cluster. Its value must be the same across all instances in
  the cluster. The cluster name is displayed in `box.info.cluster.name` (gh-5029).
```

Reviewer: Phrases "the entire cluster with all its replicasets" and "all instances of all the replicasets in one cluster" contain repetitions of the same thought, so I suggest shortening this text.

> * New option `box.cfg.replicaset_name` allows to assign a human-readable name to
>   the replicaset. Works the same as `box.cfg.replicaset_uuid`. It has to match
>   in all instances of one replicaset and is displayed in
>   `box.info.replicaset.name` (gh-5029).

Reviewer: ```suggestion
* A new option `box.cfg.replicaset_name` allows assigning a human-readable name
  to a replicaset. It works the same as `box.cfg.replicaset_uuid`. Its value
  must be the same across all instances of one replicaset. The replicaset name
  is displayed in `box.info.replicaset.name` (gh-5029).
```

Author: Done.

> * New option `box.cfg.instance_name` allows to assign a human-readable name to
>   one instance. Works the same as `box.cfg.instance_uuid`. The name has to be
>   unique in its replicaset. It is displayed in `box.info.name`. Names of the
>   other replicas in the same replicaset are visible in
>   `box.info.replication[id].name` (gh-5029).

Reviewer: ```suggestion

* A new option `box.cfg.instance_name` allows assigning a human-readable name
  to an instance. Works the same as `box.cfg.instance_uuid`. Its value must be
  unique in its replicaset. The instance name is displayed in `box.info.name`.
  Names of other replicas in the same replicaset are visible in 
  `box.info.replication[id].name` (gh-5029).
```

Author: Done.

  the entire cluster with all its replicasets. It has to match in all instances
  of all the replicasets in one cluster and is displayed in
  `box.info.cluster.name` (gh-5029).

* New option `box.cfg.replicaset_name` allows to assign a human-readable name to
  the replicaset. Works the same as `box.cfg.replicaset_uuid`. It has to match
  in all instances of one replicaset and is displayed in

>   `box.info.replicaset.name` (gh-5029).

Author: I didn't hear about this idea anything until you mentioned it here. At the same time I don't see why would it even be needed.

>   `box.info.replicaset.name` (gh-5029).

Reviewer 2: > I heard an idea about generating instance/replicaset UUID based on the instance/replicaset name (hashing alike). I guess, from @Mons. Is it declined? If so, why?

I think you're mentioning @ochaton's script which turns a readable string into a uuid (like l33tsp34k but with chars allowed for a uuid). It seems this was merely a workaround, and isn't needed anymore since we have normal instance names.

>   `box.info.replicaset.name` (gh-5029).

Reviewer 3: Well, ok. Just noted.
